src/components/elements/CommonMenuButton.tsx

```diff
@@ -105,49 +114,84 @@ const CommonMenuButton = ({
         // Bug: Opening the CommonMenu applies padding to the body, which can look weird on mobile.
         // It's sort of fixable with disableScrollLock, but that seems to introduce other scroll problems.
         // disableScrollLock={true}
-        {...MenuProps}
+        {...menuProps}
       >
         {items.map(
           ({
             key,
             to,
             title,
+            Icon,
             divider,
             onClick,
             disabled,
+            disabledReason,
             ariaLabel,
             openInNew,
             linkState,
-          }) =>
-            divider ? (
-              <Divider key={key} />
-            ) : to ? (
-              <MenuItem
-                key={key}
-                component={RouterLink}
-                to={to}
-                aria-label={ariaLabel}
-                openInNew={openInNew}
-                state={linkState}
-              >
-                {title}
-              </MenuItem>
-            ) : (
+          }) => {
+            if (divider) return <Divider key={key} />;
+
+            const props = {
+              key,
+              'aria-label': ariaLabel,
+              disabled,
+            };
+
+            const menuItemLabel = (
+              <Stack direction='row'>
+                {Icon && (
+                  <ListItemIcon>
+                    <Icon />
+                  </ListItemIcon>
+                )}
+                <Stack direction='column'>
+                  {title}
+                  {disabled && disabledReason && (
+                    // We often use tooltips to indicate the reason something is disabled,
```

Comment:
Following up from our conversation in the design session about adding disabled reason as a tooltip on action menu items: This turns out to be a bit annoying, since `MenuItem` has to be the direct child of `Menu`, so we can’t wrap it in a tooltip. We could probably create our own disabling mechanism with buttons, but to me, this sounds a bit tricky to get right. So I’m proposing instead a low lift solution here. Let me know if you think this looks good-enough-for-now:

![Screenshot 2025-02-05 at 4 07 23 PM](https://github.com/user-attachments/assets/35c28b5e-10de-4aae-8de0-c9db8ae69d13)

(Note that I’ve removed “Dense” for this table in particular, which is inconsistent with other table row actions, but I think looks better with disabled reasons)

Reply:
This looks good enough to me, thanks!